Assets/Scripts/Interactions/Mushrooms/DeathMushroom.cs

```diff
@@ -0,0 +1,7 @@
+﻿public class DeathMushroom : MotherMushroom
+{
+    protected override void StartEffect()
+    {
+        GameManager.Instance.Die("You ate an uneatable Mushroom! Please try again!");
```

Comment:
Can't we just call Gamemanager.Instance.Die()

Reply:
We could, but `DeathMushroom` inherits from `MotherMushroom` which handles all the logic for when to call this method. This way Code duplicates will likely not occur.